cram/cram_codecs.c

```diff
@@ -1630,6 +1611,18 @@ cram_codec *cram_byte_array_stop_decode_init(char *data, int size,
 	return NULL;
 
     c->codec  = E_BYTE_ARRAY_STOP;
+    switch (option) {
+    case E_BYTE_ARRAY_BLOCK:
+        c->decode = cram_byte_array_stop_decode_block;
+	break;
+    case E_BYTE_ARRAY:
+        c->decode = cram_byte_array_stop_decode_char;
+	break;
+    default:
+        fprintf(stderr, "byte_array_stop codec only supports BYTE_ARRAYs.\n");
+        free(c);
+        return NULL;
+    }
     c->decode = (option == E_BYTE_ARRAY_BLOCK)
```

Comment:
Duplicate code to the above new switch statement. Please delete one or the other.